build/jqassistant/rules/rules.xml

```diff
@@ -600,6 +601,7 @@
                 AND NOT misnamedType:Anonymous
                 AND NOT misnamedType:Deprecated
                 AND NOT externalParentType:Deprecated
+                AND NOT externalParentType:SuppressJQAssistant
```

Comment:
> Yeah, don't do that :)

ohh so this will go away then 🙈

Reply:
Well this can stay, but the `@SuppressJQAssistant` annotation you probably added should go away.

Not sure why `BindingContext` would extend `NamedValues`... the whole point what to get people to write `ctx.params().get("something")`, wasn't it? `ctx.get("something")` could get confusing.

Reply:
For now, we only have the named values in the `withParameters(...)` and there:
https://github.com/hibernate/hibernate-search/pull/4061#discussion_r1540942440

soooo that's why I thought the idea was to do the same thing here (don't add additional params but just get methods). 🙈 

as for the suppressing annotation -- it was there already as we want the get/getoptional methods and JQ won't let us do that on "public API classes" 🙈.
I've updated the code to have the params method.